### PR TITLE
reduce object allocation

### PR DESCRIPTION
**Pull request by markgeek**
**before**

```
 →  stackprof stackprof-object-20029-1436775798.dump --text
==================================
  Mode: object(5000)
  Samples: 3005 (0.00% miss rate)
  GC: 0 (0.00%)
==================================
     TOTAL    (pct)     SAMPLES    (pct)     FRAME
      1680  (55.9%)         840  (28.0%)     Grape::Util::StackableValues#[]
       415  (13.8%)         390  (13.0%)     block in Grape::Request#headers
       590  (19.6%)         175   (5.8%)     Grape::Request#headers
       250   (8.3%)         125   (4.2%)     Grape::Util::InheritableValues#merge
        75   (2.5%)          75   (2.5%)     ActiveSupport::HashWithIndifferentAccess#convert_key
       225   (7.5%)          75   (2.5%)     block in ActiveSupport::HashWithIndifferentAccess#update
       100   (3.3%)          75   (2.5%)     Grape::Util::InheritableValues#keys
        75   (2.5%)          75   (2.5%)     block in Grape::Middleware::Base#mime_types
        75   (2.5%)          75   (2.5%)     ActiveSupport::HashWithIndifferentAccess#convert_value
```

**after**

```
 →  stackprof stackprof-object-19555-1436772648.dump --text
==================================
  Mode: object(5000)
  Samples: 2530 (0.00% miss rate)
  GC: 0 (0.00%)
==================================
     TOTAL    (pct)     SAMPLES    (pct)     FRAME
      1260  (49.8%)         630  (24.9%)     Grape::Util::StackableValues#[]
       335  (13.2%)         175   (6.9%)     Grape::Request#headers
       160   (6.3%)         160   (6.3%)     block in Grape::Request#headers
       250   (9.9%)         125   (4.9%)     Grape::Util::InheritableValues#merge
       225   (8.9%)          75   (3.0%)     block in ActiveSupport::HashWithIndifferentAccess#update
        75   (3.0%)          75   (3.0%)     ActiveSupport::HashWithIndifferentAccess#convert_key
        75   (3.0%)          75   (3.0%)     block in Grape::Middleware::Base#mime_types
        95   (3.8%)          75   (3.0%)     Rack::Utils#parse_nested_query
       100   (4.0%)          75   (3.0%)     Grape::Util::InheritableValues#keys
        75   (3.0%)          75   (3.0%)     ActiveSupport::HashWithIndifferentAccess#convert_value
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Next Release
 * [#1045](https://github.com/intridea/grape/pull/1045): Do not convert `Rack::Response` to `Rack::Response` in middleware - [@dmitry](https://github.com/dmitry).
 * [#1048](https://github.com/intridea/grape/pull/1048): Only dup `InheritableValues`, remove support for `deep_dup` - [@toddmazierski](https://github.com/toddmazierski/)
 * [#1052](https://github.com/intridea/grape/pull/1052): Reset `description[:params]` when resetting validations - [@marshall-lee](https://github.com/marshall-lee).
+* [#1065](https://github.com/intridea/grape/pull/1065): Avoid object allocation in util/stackable_values.rb and http/request.rb - [@markgeek](https://github.com/markgeek)
```
Comment (dblock) [inline CHANGELOG.md]: This is a bit of an internal thing, I don't think we need a changelog line. If you think we do I am totally ok leaving it, but then make it look like other changelog entries (change util/stackable_values.rb to the class name and add a period at the end). Also, it's not like StackableValues avoids allocation altogether, I think it should say "Reduce object allocation ...".

**Comment by dblock** [comment]
Thanks. See my comment above and please squash your commits.

**Comment by markgeek** [comment]
I have removed the commit which updated the changelog.

**Comment by dblock** [comment]
Merged via https://github.com/intridea/grape/commit/018941edee3797f377b2288f95d5da358167c00a.

You might also want to see whether calling `.freeze` on the constant strings does anything to the allocation pattern or perf.

**Comment by markgeek** [comment]
OK

**Pull request closed without merging**